src/main/java/com/arkvis/orchid/PeriodCalendar.java

```diff
@@ -21,4 +24,44 @@ public void addMenstruation(LocalDate date) {
         day.addMenstruation();
         dayMap.put(date, day);
     }
+
+    public LocalDate getNextMenstruationDate() {
+        List<Day> cycleStartDays = getCycleStartDays();
+        if (cycleStartDays.isEmpty()) return null;
+
+        Day lastCycleStartDay = getLastItem(cycleStartDays);
+        long averageCycleLength = getAverageCycleLength(cycleStartDays);
+        return lastCycleStartDay.getDate().plusDays(averageCycleLength);
+    }
+
+    private long getAverageCycleLength(List<Day> cycleStartDays) {
```

Comment:
Can we use an object like "PeriodPredictor" to encapsulate the  prediction functionality?